corehq/apps/app_manager/models.py

```diff
@@ -5996,9 +5996,17 @@ def update_status(cls, domain, app_id, build_id, location_id, version, status):
         :param version: version number
         :param status: expected status
         """
-        current_release, created = cls.objects.get_or_create(
-            domain=domain, app_id=app_id, build_id=build_id, location_id=location_id, version=version)
-        current_release.activate() if status else current_release.deactivate()
+        try:
+            enabled_app_release = LatestEnabledAppReleases.objects.get(
+                domain=domain, app_id=app_id, build_id=build_id, location_id=location_id, version=version
+            )
+        except cls.DoesNotExist:
+            enabled_app_release = LatestEnabledAppReleases(
+                domain=domain, app_id=app_id, build_id=build_id, location_id=location_id, version=version
+            )
+        enabled_app_release.active = status
```

Comment:
I would call the argument to this method `active` instead of `status`, since that is the field you are setting. `status=True` doesn't really indicate anything.